front/lib/api/assistant/conversation.ts

```diff
@@ -1470,6 +1519,11 @@ export async function* retryAgentMessage(
   void
 > {
   class AgentMessageError extends Error {}
+
+  if (!canReadMessage(auth, message)) {
+    throw new AgentMessageError("Agent cannot be used by this user");
```

Comment:
IMHO, we should not throw for permissions reasons around conversation since this resource can be easily shared, we should expect this behavior to happen.